src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -969,25 +969,23 @@ void Lowering::LowerHWIntrinsicCreate(GenTreeHWIntrinsic* node)
 
         assert((simdSize == 8) || (simdSize == 16));
 
-        if ((argCnt == 1) || (simdSize == 8) || (vecCns.i64[0] == vecCns.i64[1]))
+        if (VectorConstantIsBroadcastedI64(vecCns, (argCnt == 1) ? argCnt : (simdSize / 8)))
```

Comment:
Nit:
```suggestion
        if ((argCnt == 1) || VectorConstantIsBroadcastedI64(vecCns, simdSize / 8))
```

would it be wrong to always call `VectorConstantIsBroadcastedI64(vecCns, simdSize / 8)` for simplicity?